lib/ansible/modules/cloud/vmware/vmware_dvswitch_lacp.py

```diff
@@ -0,0 +1,402 @@
+# -*- coding: utf-8 -*-
+#
+# Copyright: (c) 2018, Christian Kotte <christian.kotte@gmx.de>
+#
+# GNU General Public License v3.0+ (see COPYING or https://www.gnu.org/licenses/gpl-3.0.txt)
+
+from __future__ import absolute_import, division, print_function
+__metaclass__ = type
+
+ANSIBLE_METADATA = {
+    'metadata_version': '1.1',
+    'status': ['preview'],
+    'supported_by': 'community'
+}
+
+DOCUMENTATION = '''
+---
+module: vmware_dvswitch_lacp
+short_description: Manage LACP configuration on a distributed switch
+description:
+    - This module can be used to configure Link Aggregation Control Protocol (LACP) support mode and Link Aggregation Groups (LAGs).
+version_added: 2.8
+author:
+- Christian Kotte (@ckotte)
+notes:
+    - Tested on vSphere 6.7
+requirements:
+    - "python >= 2.6"
+    - PyVmomi
+options:
+    switch:
+      description:
+      - The name of the distribute vSwitch to manage.
+      required: True
+      aliases: ['dvswitch']
+    support_mode:
+        description:
+            - The LACP support mode.
+            - 'C(basic): One Link Aggregation Control Protocol group in the switch (singleLag).'
+            - 'C(enhanced): Multiple Link Aggregation Control Protocol groups in the switch (multipleLag).'
+        type: str
+        default: 'basic'
+        choices: ['basic', 'enhanced']
+    link_aggregation_groups:
+        description:
+            - Can only be used if C(lacp_support) is set to C(enhanced).
+            - 'The following parameters are required:'
+            - '- C(name) (string): Name of the LAG.'
+            - '- C(uplink_number) (int): Number of uplinks. Can 1 to 30.'
+            - '- C(mode) (string): The negotiating state of the uplinks/ports.'
+            - '   - choices: [ active, passive ]'
+            - '- C(load_balancing_mode) (string): Load balancing algorithm.'
+            - '   - Valid attributes are:'
+            - '   - srcTcpUdpPort: Source TCP/UDP port number.'
+            - '   - srcDestIpTcpUdpPortVlan: Source and destination IP, source and destination TCP/UDP port number and VLAN.'
+            - '   - srcIpVlan: Source IP and VLAN.'
+            - '   - srcDestTcpUdpPort: Source and destination TCP/UDP port number.'
+            - '   - srcMac: Source MAC address.'
+            - '   - destIp: Destination IP.'
+            - '   - destMac: Destination MAC address.'
+            - '   - vlan: VLAN only.'
+            - '   - srcDestIp: Source and Destination IP.'
+            - '   - srcIpTcpUdpPortVlan: Source IP, TCP/UDP port number and VLAN.'
+            - '   - srcDestIpTcpUdpPort: Source and destination IP and TCP/UDP port number.'
+            - '   - srcDestMac: Source and destination MAC address.'
+            - '   - destIpTcpUdpPort: Destination IP and TCP/UDP port number.'
+            - '   - srcPortId: Source Virtual Port Id.'
+            - '   - srcIp: Source IP.'
+            - '   - srcIpTcpUdpPort: Source IP and TCP/UDP port number.'
```

Comment:
👍

Reply:
I'm not sure how to document this properly and I'm not sure how this would finally look in the browser. I just took this from another module.